echo-pubsub-aws/src/test/java/com/netflix/spinnaker/echo/pubsub/aws/AmazonSQSSubscriberTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pubsub.aws;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import ch.qos.logback.classic.Level;
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.echo.config.AmazonPubsubProperties;
+import com.netflix.spinnaker.echo.jackson.EchoObjectMapper;
+import com.netflix.spinnaker.echo.pubsub.PubsubMessageHandler;
+import com.netflix.spinnaker.kork.aws.ARN;
+import com.netflix.spinnaker.kork.test.log.MemoryAppender;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class AmazonSQSSubscriberTest {
+
+  @Mock private AmazonSNS amazonSNS;
+
+  @Mock private AmazonSQS amazonSQS;
+
+  @Mock private PubsubMessageHandler pubsubMessageHandler;
+
+  @Mock private Registry registry;
+
+  private ARN queueARN = new ARN("arn:aws:sqs:us-west-2:100:queueName");
+  private ARN topicARN = new ARN("arn:aws:sns:us-west-2:100:topicName");
+
+  private AmazonPubsubProperties.AmazonPubsubSubscription subscription =
+      new AmazonPubsubProperties.AmazonPubsubSubscription(
+          "aws_events", topicARN.getArn(), queueARN.getArn(), "", null, null, 3600);
+
+  private ObjectMapper objectMapper = EchoObjectMapper.getInstance();
+
+  private SQSSubscriber subject;
+
+  @BeforeEach
+  void setUp() throws Exception {
+    MockitoAnnotations.openMocks(this);
```

Comment:
I recommend
```
@ExtendWith(MockitoExtension.class)
```
instead of calling openMocks directly.  If you do keep the call to openMocks, save the AutoCloseable return value  and call close on it in an `@AfterEach` method.

Reply:
Done